src/dev/build/lib/config.js

```diff
@@ -101,51 +106,32 @@ export async function getConfig({ isRelease, targetAllPlatforms, versionQualifie
       }
 
       if (process.platform === 'linux') {
-        return [this.getLinuxPlatform()];
+        return [this.getPlatform('linux', 'x64')];
       }
 
-      return [this.getPlatformForThisOs(), this.getLinuxPlatform()];
+      return [this.getPlatformForThisOs(), this.getPlatform('linux', 'x64')];
     }
 
-    /**
-     * Get the linux platform object
-     * @return {Platform}
-     */
-    getLinuxPlatform() {
-      return platforms.find((p) => p.isLinux());
-    }
-
-    /**
-     * Get the windows platform object
-     * @return {Platform}
-     */
-    getWindowsPlatform() {
-      return platforms.find((p) => p.isWindows());
-    }
-
-    /**
-     * Get the mac platform object
-     * @return {Platform}
-     */
-    getMacPlatform() {
-      return platforms.find((p) => p.isMac());
+    getPlatform(name, arch) {
+      return platforms.find((p) => {
+        return name === p.getName() && arch === p.getArchitecture();
+      });
     }
```

Comment:
The goal for the original methods was to always return a specific platform, with the new implementation we might return undefined if the arguments are wrong, so I think we should throw if the right platform can't be found.

Then we can update `getPlatformForThisOs()` to pass `os.platform()` and `os.arch()` to `getPlatform()`.